index.js

```diff
@@ -94,9 +94,9 @@ function recurse(parent, theirs, mine, path, options) {
  */
 function processKeyValuePair(key, value, parent, theirs, mine, path, options) {
   options = options || {};
-  var parentValue = typeof parent !== 'undefined' ? parent[key] : undefined;
-  var theirsValue = typeof theirs !== 'undefined' ? theirs[key] : undefined;
-  var mineValue = typeof mine !== 'undefined' ? mine[key] : undefined;
+  var parentValue = typeof parent !== 'undefined' ? (parent !== null ? parent[key] : parent ) : undefined;
```

Comment:
There is a difference because the 3-way-diff would return N instead of E.  If it's undefined it's creating a new key, if it's defined as null then it's an edit.

So you want nested if statements?